### ISSUE TITLE
Lab detail screen mockup

### ISSUE THREAD

**Issue by internetmosquito** [wizbots/labtab#7]
Design the mockup screen for the lab detail selected, taking into account tablet dimensions and size

This is an important factor to take into account, we cannot go with the current roster view that we have at Wizbots
![roster_mentor](https://cloud.githubusercontent.com/assets/1500392/19874570/93d83254-9fc6-11e6-8513-2f6933b1d58f.jpg

We need to think how to show all this data in a comprehensive manner in a tablet.

**Comment by solnishka** [comment]
@internetmosquito , @wizbots , could you please take a look if you like the idea or no:

![lab-detail](https://cloud.githubusercontent.com/assets/9883696/19964709/6e3f8140-a1d4-11e6-993e-7272d2385189.jpg)

It's the tablet screen. We have 7 levels, one is open on the right side, the other 6 are buttons, when click the button an appropriate level opens on the right and the previous opened level button appears except the clicked one.
The rest is clear I think.

**Comment by internetmosquito** [comment]
I like the idea, but I'd remove the Wizbots logo on top to add a button to access video recording easily, something like...mentor selects 2-3 kids (long taps) and then clicks on button and is redirected to the new video view.

Just one thing, everytime you click a different level on the PTM the kid list changes nope? Is it like a filter or they stay there no matter what level do we choose? I mean, do we show only kids with the selected level already?

Another thing, I was talking about a "kid detail view" in the doc, this might not be necessary if we add this on this view...something like, tap on a kid, and expand a section for that kid where mentor can see all the relevant info for that kid, tap again and it colapses....what you think? Good job by the way!

**Comment by internetmosquito** [comment]
Adding Sourabh to this so we get his feedback

**Comment by kapoorutd** [comment]
Few points:

This view is very specific for tablet and may not work well on phones.
Extreme right section (where troubleshooting, labtime and projects are written upside down) is very difficult to read. 
We should use less of images and more of color as per the latest guidelines - Thats what i feel . Please ignore if its not aligned with business  idea.

**Comment by kapoorutd** [comment]
Overall great concept but thought to share some concern as a developer.

**Comment by solnishka** [comment]
> I like the idea, but I'd remove the Wizbots logo on top to add a button to access video recording easily, something like...mentor selects 2-3 kids (long taps) and then clicks on button and is redirected to the new video view.

As I know in the current (desktop) functionality there is just a list of students - not selectable. I can be mistaken. But for now this is like report, so if you want to add anything clickable, please let me know, we may change the concept. It's not a problem to add another button. Just describe me the new functionality you'd like to implement here. Thanks.

**Comment by solnishka** [comment]
> Just one thing, everytime you click a different level on the PTM the kid list changes nope? Is it like a filter or they stay there no matter what level do we choose? I mean, do we show only kids with the selected level already?

no, the kids list is the same - it's like a group of students - one class. So before you come to the page you select one class with constant amount of children in it. As I see it, in the desktop we see 7 blocks and here we don't have such a width - so by clicking buttons we can make only one block (level) to be visible. 
So we always see the same kids even if still haven't get to the selected level but they are in the same group.

**Comment by solnishka** [comment]
>Another  thing, I was talking about a "kid detail view" in the doc, this might not be necessary if we add this on this view...something like, tap on a kid, and expand a section for that kid where mentor can see all the relevant info for that kid, tap again and it colapses....what you think? Good job by the way!

I think it's not a problem to make kids fields clickable. And I'll design the page with kid's info. The only thing - as I know modals are bad decision for the mobiles - so let's do it as a separate page from which you can come back to the main one. @kapoorutd, what do you think?

**Comment by solnishka** [comment]
> Extreme right section (where troubleshooting, labtime and projects are written upside down) is very difficult to read. 

do you have any suggestions how to improve it to make it easy for development? I'm open for discussion, just have no idea how to change the concept to leave the concept of the table. For the mentors it will be easy to read cause they already have the same view on desktop - so it's not a problem - it's like a short variant of the thing they already have.  We can remove purple statistics in the top level section if it makes the development easier. But the pics in general - mentors know them and have been using them for long - so they make it easier to find the level they look for. So if you have anything to suggest please let me know and we'll think how to improve the view.

**Comment by solnishka** [comment]
> This view is very specific for tablet and may not work well on phones.

Could you please describe me, why? Thanks.

**Comment by solnishka** [comment]
> We should use less of images and more of color as per the latest guidelines

The images here help us not to use long words and descriptions - all involved know what they mean and they help to use as small place as possible.

**Comment by solnishka** [comment]
@internetmosquito , @kapoorutd , waiting for your thoughts and suggestions to continue with this page

**Comment by kapoorutd** [comment]
@solnishka does it make sense to have a skype call so that we can collaborate better.

**Comment by internetmosquito** [comment]
@solnishka @kapoorutd creating a shared document for this because is becoming hard to follow the comments here

**Comment by internetmosquito** [comment]
@solnishka @kapoorutd here https://docs.google.com/document/d/1jlYr0ZgHVs3NanmYnJMWTWuMqnxKKcU8pYAqR_culhA/edit?usp=sharing

Have a look and let's discuss this via Skype whenever you're ready

**Comment by solnishka** [comment]
@internetmosquito , left my comments in the doc

**Comment by solnishka** [comment]
@internetmosquito, @kapoorutd, I have one more suggestion, it may be better for tablets.

The first view:

the buttons are levels filtering: when no button is clicked - general statistics is displayed in the table, if we click any button, then selected level statistics is displayed. If there is not enough vertical space, vertical scroll apears in the table  - not the whole page - only table. If possible table header has to be freeze.

![lab-detail--view-1](https://cloud.githubusercontent.com/assets/9883696/20058545/a6cf9546-a502-11e6-9503-95dbddb122e0.jpg)

When there is no buttons clicked and we click on any of the students we get the view 2:

All the levels statistics for the clicked student appears. If we also clicked any of the filtering buttons, then only selected levels will apear for the student that was selected.

![lab-detail--view-2](https://cloud.githubusercontent.com/assets/9883696/20058573/b9727ad8-a502-11e6-9376-a7d9d5eb54c4.jpg)

If we click on any level then we'll see the view 3 (some kind of the one that Alejandro suggested):

![lab-detail--view3](https://cloud.githubusercontent.com/assets/9883696/20058587/c6904c36-a502-11e6-9c35-c98e35297adc.jpg)

Here we can see student's progress on every step of the level we selected.

**Comment by internetmosquito** [comment]
@solnishka Not what I had in mind originally but I like the idea, no level selected and we get a general overview of how kids are doing, not as detailed as the desktop roster of course, but good enough I recon.

I like using buttons to filter results, this was also suggested by Sourabh, and it makes to me. My only concern is that in first and second screen you seem to suggest that we can multi-select several levels at once (which makes sense) but then if we have one kid "expanded"  and we click on more than one button what happens?

About scrolling, I agree, header must stay up if possible...I don't really think we're gonna need scrolling that much, but we better plan for it.

What about the button to record video? How are we going to show kids that are selected before taping the record video button? Something like a checked ticker on left or different for selected rows? The idea is to select one or several kids (long tap) and then click on the record video button, that way we already have the kids selected.

What about these design @kapoorutd are they easily approachable / doable?

**Comment by kapoorutd** [comment]
@solnishka <https://github.com/solnishka> @internetmosquito
<https://github.com/internetmosquito> I think we are in right direction and
now it has lot of Android widgets.

Only few comments :

1. We should not create a border in blue for Student roster as you wont
have this much spacing for Mobile phones. Please increase list row to max
width of phone.
2. shouldnt showing student progress be at next screen? As per me its
confusing n complex to do expandable list .
3. How do we edit student progress?
4. Shouldnt header be slimmed down if we want to make it stick to top. Also
as per Android guidelines hamburger icon should be on left and not to right
until and unless we are in Arabic country where users read from right to
left.
5. What are the various screen under hamburger icon?

**Comment by solnishka** [comment]
> What about the button to record video? How are we going to show kids that are selected before taping the record video button? Something like a checked ticker on left or different for selected rows? The idea is to select one or several kids (long tap) and then click on the record video button, that way we already have the kids selected.

agree, we can add one more row to the table, and one more button in filters line, all the selected students will be added to the video adding form.

![lab-detail-view-4](https://cloud.githubusercontent.com/assets/9883696/20099419/71150b08-a5c9-11e6-8c8b-5e4f102d402c.jpg)

**Comment by solnishka** [comment]
> 1. We should not create a border in blue for Student roster as you wont
have this much spacing for Mobile phones. Please increase list row to max
width of phone.

I just used our general style, if we need to remove any borders - I think it's not a problem at all.

**Comment by solnishka** [comment]
> 2. shouldnt showing student progress be at next screen? As per me its
confusing n complex to do expandable list .

it's useful if a mentor wants to compare anything. But anyway @internetmosquito  has to decide I think

**Comment by solnishka** [comment]
> . How do we edit student progress?
@internetmosquito, do we have to do it with this app? If yes, then we can add "edit" button to every student and I'll design the screen with changes - please create the ticket for this.

![lab-detail--view-5](https://cloud.githubusercontent.com/assets/9883696/20099743/3e68cb84-a5cb-11e6-8f45-e351c285c56d.jpg)

**Comment by solnishka** [comment]
> Shouldnt header be slimmed down if we want to make it stick to top. Also
as per Android guidelines hamburger icon should be on left and not to right
until and unless we are in Arabic country where users read from right to
left.

I think it's not a problem to change logo and hamburger

**Comment by kapoorutd** [comment]
I think we are very close to Android specs. If Alejandro can comment on
Edit and placement of hamburger icon then we should be good to go.
Great effort @solnishka.

**Comment by solnishka** [comment]
> What  are the various screen under hamburger icon? 

It will be a menu - all the screens we'll have I think

**Comment by internetmosquito** [comment]
@solnishka @kapoorutd Looking solid, I like the idea of the pencil to go to this "very detailed view of student" where, in the future, we will be able to probably upgrade to next level, add nuggets, wizchips, etc, for now is ok with it being just read only.

So summarising:

1.- Have a list of buttons to filter per level, one button to create video
2.- If no filter is selected we have a general overview of students (first image from Tatyana) depicting current level, projects completed, lab time, pending, done and skipped.
3. If no filter is selected and we tap on a student we show the general statistics per level for that kid (second image from Tatyana)
4.- If a level is selected (filter) then we show only kids with selected level? This was unclear from third screen...I would say YES
5.- If a level is selected and we tap on a student we seed specific stats for that level (mostly projects and status)
6.- If we long tap on a student/s they get marked (last image from Tatyana) and a pencil appears on right outmost part of the student row
7.- For selected student/s we can then click on record video and the new video screen will show up
8.- For a selected student we can click the pencil icon and go to the Student Detail view ( I think something similar to second screen, when general stats per project are shown combined with taping on each level to get more detailed info like third screen could be a good starting point)

If we all agree on this, I update the doc and create the issues for the student detail view mockups and new video screen mockup (if I haven't already)

**Comment by internetmosquito** [comment]
@solnishka Have you thought on a way to highlight kids that are about to pass next level? It would be great if we could do that, not really urgent until we can actually act on kids, but would be good to have this design already

**Comment by solnishka** [comment]
> 3. If no filter is selected and we tap on a student we show the general statistics per level for that kid (second image from Tatyana)
- we show all the levels for this kid

> 4.- If a level is selected (filter) then we show only kids with selected level? This was unclear from third screen...I would say YES
-yes

> 6.- If we long tap on a student/s they get marked (last image from Tatyana) and a pencil appears on right outmost part of the student row 
-this screen appears when we tap on the level under the student from the second view.

**Comment by solnishka** [comment]
> @solnishka Have you thought on a way to highlight kids that are about to pass next level? It would be great if we could do that, not really urgent until we can actually act on kids, but would be good to have this design already

I think we can mark the kid's name with color - e.g. purple. What do you think?

**Comment by internetmosquito** [comment]
@solnishka not sure I get this...
 "this screen appears when we tap on the level under the student from the second view." ...are you saying we need to have a filter enabled in order to be able to select kids for shooting video or edition?

About the purple, not sure if this will be clear enough...do we have room for an icon that we can put beside the kids name? something like this maybe

![icon_cup-317b1951113337e86ee20b3bb138bf7449e21b5303b3aa407603b31cd782fbdf](https://cloud.githubusercontent.com/assets/1500392/20110061/2497640a-a5e3-11e6-8dea-2dd1c6a9a469.png)

**Comment by solnishka** [comment]
> @solnishka not sure I get this...
"this screen appears when we tap on the level under the student from the second view." ...are you saying we need to have a filter enabled in order to be able to select kids for shooting video or edition?

sorry for this - it was just misunderstanding - I thought about another action when wrote this.

**Comment by solnishka** [comment]
> About the purple, not sure if this will be clear enough...do we have room for an icon that we can put beside the kids name? something like this maybe icon_cup-317b1951113337e86ee20b3bb138bf7449e21b5303b3aa407603b31cd782fbdf

I think we have to ask @kapoorutd about the space - from me it's not a problem, but if we have enough horizontal space or no? @kapoorutd , what do you think?

**Comment by solnishka** [comment]
I have an idea - will provide the mockup soon.

**Comment by solnishka** [comment]
![lab-detail---3](https://cloud.githubusercontent.com/assets/9883696/20140671/71c37e20-a69d-11e6-807c-44367ba8ece5.jpg)

**Comment by internetmosquito** [comment]
@solnishka Like it, looks great!

**Comment by solnishka** [comment]
![21 13331](https://cloud.githubusercontent.com/assets/9883696/20485972/eca7a1e6-b00e-11e6-83f4-5127d85facd3.png)

**Comment by solnishka** [comment]
Now when we want to add new video we check the students and click Video btn. When we click View btn we go to page:
![21 14441](https://cloud.githubusercontent.com/assets/9883696/20486056/4bca6e4c-b00f-11e6-987a-d393e0a66863.png)
if we click Edit btn we go to page:
![21 1111](https://cloud.githubusercontent.com/assets/9883696/20486105/8a02f4fe-b00f-11e6-9c7f-b9aa3f3aef3f.png)
